python/lance/types/base.py

```diff
@@ -0,0 +1,26 @@
+#  Copyright (c) 2022. Lance Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import platform
+from abc import ABC
+
+import pyarrow as pa
+
+
+class LanceType(pa.ExtensionType, ABC):
+    def __init__(self, storage_type, extension_name):
+        if platform.system() != "Linux":
```

Comment:
Should we just do one system check either here or `__init__.py`?

Reply:
fixed